client/actions/events/api.ts

```diff
@@ -547,6 +549,89 @@ const uploadFiles = (event) => (
     }
 );
 
+function updateLinkedPlanningsForEvent(
+    eventId: IEventItem['_id'],
+
+    /**
+     * these must be final values
+     * missing items will be linked, extra items unlinked
+     */
+    associatedPlannings: Array<IPlanningItem>,
+):Promise<void> {
+    return planningApi.events.getLinkedPlanningItems(eventId).then((currentlyLinked) => {
+        const currentLinkedIds = new Set(currentlyLinked.map((item) => item._id));
+
+        const toLink: Array<IPlanningItem> =
+            associatedPlannings.filter(({_id}) => currentLinkedIds.has(_id) !== true);
+
+        const toUnlink: Array<IPlanningItem> = currentlyLinked
+            .filter((item) => {
+                const createdAt = new Date(item._created);
+                const now = new Date();
+
+                const ageSeconds = (now.getTime() - createdAt.getTime()) / 1000;
+                const tooRecent = ageSeconds > 30;
+
+                if (tooRecent) {
```

Comment:
@MarkLark86 any idea how to avoid this hack?

Reply:
Only thing that comes to mind is if you also pass in the original associatedPlanning IDs. Check non-temp IDs (aka not "fake ID") that are in original that are not in updates are to be unlinked.